### PR TITLE
Merge Cadence 1.0 feature branch to master

### PR DESCRIPTION
**Pull request by turbolent**
Almost a year and almost a thousand commits later, we're also finally ready to merge the Cadence 1.0 feature branch here in flow-go 🎉

NOTE: Once this this PR got merged, flow-go will **only** support Cadence 1.0, there is no backward-compatibility for <1.0. Deploying these changes to networks running <1.0 requires a state migration!

### CODE REVIEW

**Review by SupunS — APPROVED** [review]
👏  🎉

File: fvm/environment/meter.go [hunk 1]
```diff
@@ -35,8 +35,8 @@ const (
 	ComputationKindRemoveAccountContractCode
 	ComputationKindResolveLocation
 	ComputationKindRevokeAccountKey
-	ComputationKindRevokeEncodedAccountKey
-	_
```
Comment (ramtinms) [inline fvm/environment/meter.go:39]: Is Encoded Account keys not something supported anymore?

File: fvm/environment/account_key_updater.go [hunk 2]
```diff
@@ -81,28 +81,6 @@ func NewAccountPublicKey(publicKey *runtime.PublicKey,
 // Note that scripts cannot modify account keys, but must expose the API in
 // compliance with the runtime environment interface.
 type AccountKeyUpdater interface {
```
Comment (ramtinms) [inline fvm/environment/account_key_updater.go:83]: @janezpodhostnik are these changes expected?

File: engine/execution/computation/manager.go [hunk 3]
```diff
@@ -231,12 +231,9 @@ func DefaultFVMOptions(chainID flow.ChainID, cadenceTracing bool, extensiveTraci
 			reusableRuntime.NewReusableCadenceRuntimePool(
 				ReusableCadenceRuntimePoolSize,
 				runtime.Config{
-					TracingEnabled:        cadenceTracing,
-					AccountLinkingEnabled: true,
```
Comment (ramtinms) [inline engine/execution/computation/manager.go:235]: is account linking enabled by default going forward?

**Comment by ramtinms** [comment]
For the migration of states, do we re-adjust the minimum flow balance needed for accounts, given the size of the storage is going to change for each account?

**Review by ramtinms — APPROVED** [review]
🚀 😸  Exciting!

ps: 
Just put some questions in the code to make sure all changes are expected.

**Review by peterargue — APPROVED** [review]
🎉

File: engine/execution/computation/manager.go (hunk 3 above)
Comment (turbolent) [reply engine/execution/computation/manager.go:235]: Account linking got replaced by account capability controllers, which are not behind behind a feature flag and always enabled.

File: fvm/environment/meter.go (hunk 1 above)
Comment (turbolent) [reply fvm/environment/meter.go:39]: The functions to add account keys via "encoded keys" has been deprecated when the object-oriented key management functions got added, a couple years ago. The deprecated functions  got finally removed.

**Comment by turbolent** [comment]
@ramtinms 
> For the migration of states, do we re-adjust the minimum flow balance needed for accounts, given the size of the storage is going to change for each account?

I don't know about this, probably a question for @onflow/flow-cadence-execution

**Review by jordanschalm — APPROVED** [review]
Looked at bootstrapping, model changes, and DKG and epoch test changes.

**Review by janezpodhostnik — APPROVED** [review]
Looked at FVM and migrations. LGTM!

File: fvm/environment/account_key_updater.go (hunk 2 above)
Comment (turbolent) [reply fvm/environment/account_key_updater.go:83]: Correct, see https://github.com/onflow/flow-go/pull/5856#discussion_r1591688491

**Pull request merged**